protocol-documentation.md

```diff
@@ -0,0 +1,122 @@
+# Protocol documentation
+Early version of the protocol used for communication, subject to be changed and expanded. Not all messages are implemented.
+
+# Format of messages
+
+Every message that is sent between the client and the server must be in the format described below. The `type` field describes what type of request/response is being sent, and the `payload` value contains whatever data is associated with that message. Important to note here is that payload can have many different types, i.e it can be an object, array or a specific value. Please look at the specification for each individual message to know what data the payload contains.
+
+```json
+{
+    "type": "<unique description of request/response>",
+    "payload": {
+        "data": []
+    }
+}
+```
+
+# Client messages
+Messages that are sent from a client to the server.
+
+When the position or zoom on the client's map is changed, this message should be sent to the server so that only information about busses that can be seen are sent to the client.
+```json
+{
+    "type": "geo-position-update",
+    "payload": {
+        "radius": 10,
+        "position": {
```

Comment:
I suggest using GeoJSON format for the coordinates to make our lives easier later:

location: { type: "Point", coordinates : [longitude, latitude]}

Reply:
👍 

I suppose it would also be good to rename `radius` into `maxDistance`, since that is the name for the MongoDB parameter.